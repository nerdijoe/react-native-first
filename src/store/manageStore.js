import { createStore, applyMiddleware, compose } from 'redux'
import logger from 'redux-logger'
import thunk from 'redux-thunk'

import clasApp from '../reducers'

const middlewares = applyMiddleware( logger, thunk )

const store = createStore(clasApp, middlewares)

export default store
