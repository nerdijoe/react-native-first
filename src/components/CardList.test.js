import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'

import ConnectedCardList, {CardList} from './CardList'
import store from '../store/manageStore'

describe('CardList', () => {
  let cards = [
    {
      "_id":"58d176a56d804500104b1248",
      "idName":"arrows",
      "rarity":"Common",
      "type":"Spell",
      "name":"Arrows",
      "description":"Arrows pepper a large area, damaging everyone hit. Reduced damage to Crown Towers.",
      "arena":0,
      "elixirCost":3,
      "order":1,
      "__v":0
    },
    {"_id":"58d176a56d804500104b124e","idName":"bomber","rarity":"Common","type":"Troop","name":"Bomber","description":"Small, lightly protected skeleton that throws bombs. Deals area damage that can wipe out a swarm of enemies.","arena":0,"elixirCost":3,"order":2,"__v":0},
    {"_id":"58d176a56d804500104b125b","idName":"archers","rarity":"Common","type":"Troop","name":"Archers","description":"A pair of unarmored ranged attackers. They'll help you take down ground and air units, but you're on your own with hair coloring advice.","arena":0,"elixirCost":3,"order":3,"__v":0},
    {"_id":"58d176a56d804500104b125d","idName":"knight","rarity":"Common","type":"Troop","name":"Knight","description":"A tough melee fighter. The Barbarian's handsome, cultured cousin. Rumor has it that he was knighted based on the sheer awesomeness of his mustache alone.","arena":0,"elixirCost":3,"order":4,"__v":0},
    {"_id":"58d176a56d804500104b1257","idName":"fireball","rarity":"Rare","type":"Spell","name":"Fireball","description":"Annnnnd... Fireball. Incinerates a small area, dealing high damage. Reduced damage to Crown Towers.","arena":0,"elixirCost":4,"order":5,"__v":0},
    {"_id":"58d176a56d804500104b1253","idName":"mini-pekka","rarity":"Rare","type":"Troop","name":"Mini P.E.K.K.A.","description":"The Arena is a certified butterfly-free zone. No distractions for P.E.K.K.A., only destruction.","arena":0,"elixirCost":4,"order":6,"__v":0},
  ]

  it('renders without crashing', ()=> {
    // const tree = renderer.create(<Provider store={store}><CardList cards={cards}/></Provider>)
    const tree = renderer.create(<CardList cards={cards}/>)
    expect(tree).toMatchSnapshot()
  })
})
