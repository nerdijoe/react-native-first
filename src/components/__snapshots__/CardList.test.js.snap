// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "space-around",
            "marginBottom": 10,
            "marginTop": 40,
            "padding": 20,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flexDirection": "column",
              "margin": 3,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={
              Object {
                "uri": "http://www.clashapi.xyz/images/cards/arrows.png",
              }
            }
            style={
              Object {
                "height": 65.75,
                "width": 53,
              }
            }
          />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flexDirection": "column",
              "margin": 3,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={
              Object {
                "uri": "http://www.clashapi.xyz/images/cards/bomber.png",
              }
            }
            style={
              Object {
                "height": 65.75,
                "width": 53,
              }
            }
          />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flexDirection": "column",
              "margin": 3,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={
              Object {
                "uri": "http://www.clashapi.xyz/images/cards/archers.png",
              }
            }
            style={
              Object {
                "height": 65.75,
                "width": 53,
              }
            }
          />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flexDirection": "column",
              "margin": 3,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={
              Object {
                "uri": "http://www.clashapi.xyz/images/cards/knight.png",
              }
            }
            style={
              Object {
                "height": 65.75,
                "width": 53,
              }
            }
          />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flexDirection": "column",
              "margin": 3,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={
              Object {
                "uri": "http://www.clashapi.xyz/images/cards/fireball.png",
              }
            }
            style={
              Object {
                "height": 65.75,
                "width": 53,
              }
            }
          />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flexDirection": "column",
              "margin": 3,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={
              Object {
                "uri": "http://www.clashapi.xyz/images/cards/mini-pekka.png",
              }
            }
            style={
              Object {
                "height": 65.75,
                "width": 53,
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
